internal/gatewayapi/status/gateway.go

```diff
@@ -96,8 +96,15 @@ func UpdateGatewayStatusProgrammedCondition(gw *gwapiv1.Gateway, svc *corev1.Ser
 	} else {
 		gw.Status.Addresses = nil
 	}
+
 	// Update the programmed condition.
 	gw.Status.Conditions = MergeConditions(gw.Status.Conditions, computeGatewayProgrammedCondition(gw, deployment))
+
+	if len(gw.Status.Addresses) > 16 {
```

Comment:
can we move the truncation and the status message in one place to easier to reason ?